runtime/src/main/java/org/corfudb/runtime/Backup.java

```diff
@@ -156,33 +171,40 @@ private void backup() throws IOException {
      * If the log is trimmed at timestamp, the backupTable will fail.
      * If the table has no data to be backed up, it will create a file with empty contents.
      *
-     * @param fileName   - the name of the backup file
+     * @param filePath   - the path of the backup file
      * @param uuid       - the uuid of the table which is being backed up
      * @throws IOException
      */
-    private void backupTable(String fileName, UUID uuid) throws IOException {
-        try (FileOutputStream fileOutput = new FileOutputStream(fileName)) {
+    private void backupTable(Path filePath, UUID uuid) throws IOException {
```

Comment:
Why not throw something more application-specific? Like BackupException?

Reply:
Good point. I will have discussion with Pavan and Graham about exception handling.